src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -241,41 +241,52 @@ internal partial class CodeFixService : ICodeFixService
         }
 
         public async Task<CodeFixCollection?> GetDocumentFixAllForIdInSpanAsync(
-            Document document, TextSpan range, string diagnosticId, CodeActionOptions options, CancellationToken cancellationToken)
+            Document document, TextSpan range, string diagnosticId, DiagnosticSeverity minimumSeverity, CodeActionOptions options, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            var diagnostics = (await _diagnosticService.GetDiagnosticsForSpanAsync(document, range, diagnosticId, includeSuppressedDiagnostics: false, cancellationToken: cancellationToken).ConfigureAwait(false)).ToList();
-            if (diagnostics.Count == 0)
+            var diagnostics = await _diagnosticService.GetDiagnosticsForSpanAsync(
+                document, range, diagnosticId, includeSuppressedDiagnostics: false, cancellationToken: cancellationToken).ConfigureAwait(false);
+            diagnostics = diagnostics.WhereAsArray(d => d.Severity.GreaterThanOrEqualTo(minimumSeverity));
+            if (!diagnostics.Any())
                 return null;
 
             using var resultDisposer = ArrayBuilder<CodeFixCollection>.GetInstance(out var result);
             var spanToDiagnostics = new SortedDictionary<TextSpan, List<DiagnosticData>>
             {
-                { range, diagnostics },
+                { range, diagnostics.ToList() },
             };
 
             await foreach (var collection in StreamFixesAsync(
                 document, spanToDiagnostics, fixAllForInSpan: true, CodeActionRequestPriority.None,
-                options, addOperationScope: static _ => null, cancellationToken).ConfigureAwait(false))
+                options, addOperationScope: static _ => null, cancellationToken).WithCancellation(cancellationToken).ConfigureAwait(false))
```

Comment:
why the WithCancellation?  StreamFixesAsync already marks its CT as [EnumeratorCancellation] so this should not be necessary.

Reply:
I thought both were necessary?

Reply:
from: 

> https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/async-streams#:~:text=The%20%5BEnumeratorCancellation%5D%20attribute%20is%20used%20for%20this%20purpose.,of%20the%20value%20originally%20passed%20for%20the%20parameter.

> Consider `IAsyncEnumerable<int> GetData([EnumeratorCancellation] CancellationToken token = default)`. The implementer of this method can simply use the parameter in the method body. The consumer can use either consumption patterns above:
>
> if you use `GetData(token)`, then the token is saved into the async-enumerable and will be used in iteration,

This is the scenario in teh code afaict.  So .WithCancellation should not be necessary.

Reply:
There are two scenarios:
- you make your own IAsyncEnumerable instance by calling `GetData`: put `[EnumeratorCancellation] CancellationToken token = default` in `GetData`, use `token` in the body of `GetData` and you're all set
- you get a enumerable from someone else, or one that has the wrong cancellation token baked in: use `.WithCancellation` to stamp/merge your own cancellation token onto this existing enumerable

This area is a bit confusing. I have a write-up if that helps: http://blog.monstuff.com/archives/2019/03/async-enumerables-with-cancellation.html

Reply:
https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQAYACdbAFgG4DizsA6AJQFcYAXKMAUwYGEA9mAAOUADbcATgGUpANygBjbgGcahUuQYAVABaTuEBLADm6uuQCs62gGYymEtoCew7gDkIPEgG8CJALJ7dAAOMgA2EgBBAAo+CBhlMTEITgEYbQEAa24YEkUEpJS0jOzcgEpff0Ca9ABOEgAzAQMIRV0SGLkISRI5ElgSAFkYgsTuZNSodMycmHLKnxIAX2qA1fw1oLIw9EiAITjCieLp0rn848mS2Yqq/BrahubW9s7u3v7B91Gr05myvNFisthstuhgrtInwjuNrmdbnkxkUpgC5ostk8mi1DG8uj0+gM8iNkSdUecKgwAOpQdi6eJw/4wX6M8mIha+EEPQJg7kBCE7CIkAAisJRN0Bl1ZEvR90egXq2NeHXxnyJJB+pPhaMpNLpDPFZxZhp1QM5GxqvNqkJIAEkoipnIkAKIwFg8SQQYASAA8sHYAD5hmKyTLclKTRT5nL5WQGnsGABNKATBAxcrqWP0bAkYCtLKZnkEcE2+2Ol1uj1e33+oM/ADarvdUlSLQNobOAF0SO3tVGIx3TRi+Y9FQnk6n04X5eQc3nDAXQQRlkA

Reply:
@jcouv I'm really not understanding.  The docs literally say:

> if you use GetData(token), then the token is saved into the async-enumerable and will be used in iteration,

That's exactly the case here.   The docs say: 

>  The consumer can use either consumption patterns above:

But it does not appear to be taht way in practice?

Reply:
Sorry, my answer was general (looking at the comment thread, not the code change).
If `StreamFixesAsync` has the `[EnumeratorCancellation]` attribute on its cancellation token parameter, then it bakes a token into the enumerable. That is indeed situation (1) that I described above. No `.WithCancellation` needed in that case.

Reply:
> Compare the calls to GetAsyncEnumerator in <B>d__1 and <D>d__1 to see how both are required.

@sharwell When using `[EnumeratorCancellation]` the `GetEnumerator` call from `await foreach` will have no cancellation token passed in. But the state machine will take the cancellation token given as parameter of the async-iterator method and create a merged cancellation tokens (one from `GetEnumerator` and one given as parameter) which will be available in your method body.